packages/schema/src/config/webpack.ts

```diff
@@ -197,15 +197,20 @@ export default defineUntypedSchema({
        * @type {Partial<typeof import('vue-loader')['VueLoaderOptions']>}
        */
       vue: {
-        transformAssetUrls: {
-          video: 'src',
-          source: 'src',
-          object: 'src',
-          embed: 'src'
-        },
-        compilerOptions: { $resolve: async (val, get) => val ?? (await get('vue.compilerOptions')) },
-        propsDestructure: { $resolve: async (val, get) => val ?? Boolean(await get('vue.propsDestructure')) },
-        defineModel: { $resolve: async (val, get) => val ?? Boolean(await get('vue.defineModel')) }
+        $resolve: async (val, get) => {
+          const vue = await get('vue')
+          return defu(val, {
+            transformAssetUrls: {
+              video: 'src',
+              source: 'src',
+              object: 'src',
+              embed: 'src'
+            },
+            compilerOptions: vue.compilerOptions,
+            propsDestructure: Boolean(vue.propsDestructure),
+            defineModel: Boolean(vue.defineModel)
+          })
+        }
```

Comment:
@danielroe would a change like this make sense?

Reply:
I can also just wrap everything in a big defu object, like so:
```ts
export default defineUntypedSchema({
    webpack: {
        $resolve: async (val, get) => {
            const webpack = await get('webpack')
            return defu(val, { /* all the properties in here */ })
        }
    }
})
```
Do you think that would be overkill?

Reply:
Alright, I'll revert it then. Do you have any guidelines for when it's better to make it like this? (For example, the app section in common.ts is resolved the same way as the webpack change I suggested)

Reply:
I have reverted the update, which means I *think* it is ready for review.